diesel/src/macros/mod.rs

```diff
@@ -613,12 +613,7 @@ macro_rules! __diesel_table_impl {
                 Table,
                 JoinTo,
             };
-            use $crate::associations::HasTable;
-            use $crate::insertable::Insertable;
-            use $crate::query_builder::*;
-            use $crate::query_builder::nodes::Identifier;
-            use $crate::query_source::{AppearsInFromClause, Once, Never};
-            use $crate::query_source::joins::{Join, JoinOn};
+            use $crate::query_builder::AsQuery;
```

Comment:
Any reason why those imports need to stay there?

Reply:
That's why `as_query` defined in `AsQuery` is used in the following lines. However, since we don't need to import `AsQuery` globally, I've imported it in functions that need it.

https://github.com/diesel-rs/diesel/blob/d7ce43d2130bca434cd1acc2943a0c944d1deba4/diesel/src/macros/mod.rs#L713

https://github.com/diesel-rs/diesel/blob/d7ce43d2130bca434cd1acc2943a0c944d1deba4/diesel/src/macros/mod.rs#L781

Reply:
That's a good fix. To just mention the other possible solution somewhere for the case you don't know this:

In rust it's possible to write `#crate::AsQuery::as_query(self).values()` instead, which allows to completely remove the import statement. (That's not required here, I'm just writing that as educational comment)

Reply:
That’s new for me! Thank you!